### ISSUE TITLE
Datadog timeboard does not support "log_query"

### ISSUE THREAD

**Issue by joaoe** [terraform-providers/terraform-provider-datadog#120]
In my project I'm trying to create a dashboard with a timeseries that shows number of log events groupped by event type (error, warn, info, debug). When using the datadog web interface to create the widget manually, it produces some JSON which is not compatible with the terraform datadog plugin [1].

### Terraform Version
Issue in `master`, see https://github.com/terraform-providers/terraform-provider-datadog/blob/master/datadog/resource_datadog_timeboard.go

### Affected Resource(s)
Please list the resources as a list, for example:
- datadog_timeboard

### Terraform Configuration Files
```
resource "datadog_timeboard" "example-dashboard" {
  title = "Example"
  read_only = true

  graph {
    title = "Log event counts"
    viz = "timeseries"
    request {
      log_query = {
        index = "main",
        search = {
          query = "service:my-app AND status:error"
        },
        group_by = [],
        compute = {
          aggregation = "count",
          facet = null,
          interval = "300000"
        }
      },
      aggregator = "count",
      style = {
        palette = "warm",
        type = "solid",
        width = "thick"
      }
      type = "line"
      # conditional_formats = []
    },
    autoscale = true,
    yaxis = {
      scale: "log",
      includeZero: false
    }
  }
}
```

### Expected Behavior
`log_query` should be supported 

### Actual Behavior
Use of `log_query` causes errors.

### Attachment
[1] Datadog JSON sample
```
{
  "requests": [
    {
      "log_query": {
        "index": "main",
        "search": {
          "query": "service:my-app AND status:error"
        },
        "groupBy": [],
        "compute": {
          "aggregation": "count",
          "facet": null,
          "interval": "300000"
        }
      },
      "aggregator": "count",
      "style": {
        "palette": "warm",
        "type": "solid",
        "width": "thick"
      },
      "type": "line",
      "conditional_formats": []
    }
  ],
  "viz": "timeseries",
  "autoscale": true,
  "yaxis": {
    "scale": "log",
    "includeZero": false
  },
  "status": "done"
}
```

**Comment by joaoe** [comment]
![datadogncdx](https://user-images.githubusercontent.com/461983/48702732-f089ab80-ebf1-11e8-8316-258595ff06fb.png)

**Comment by joaoe** [comment]
An alternative would be to allow `log_query` to be inserted in the terraform file as a plain string, like
```
resource "datadog_timeboard" "example-dashboard" {
  graph {
    request {
      log_query = "{json blob}"
    }
  }
}
```
So you do not have to worry about implementing the whole data structure.

**Comment by h3x89** [comment]
any plans for this issue?

**Comment by nmuesch** [comment]
Hey, @joaoe and @h3x89, 👋 Apologies for the delay here. We just released the 2.1.0 version of the Datadog Terraform Provider that includes the new Dashboard Resource. This resource supports the creation of the Log Query in a widget, here's an example:

```resource "datadog_dashboard" "log_query_dash_ex" {
  title         = "Log Query Example Dash"
  description   = "Created using the Datadog provider in Terraform"
  layout_type   = "ordered"
  is_read_only  = true

  widget {
    timeseries_definition {
      request {
        log_query {
          index = "main"
          compute = {
            aggregation = "count"
            facet = "@duration"
            interval = 5000
          }
          search = {
            query = "status:info"
          }
          group_by {
            facet = "null"
            limit = 10
            sort = {
              aggregation = "avg"
              order = "desc"
              facet = "@duration"
            }
          }
        }
        display_type = "area"
      }
    }
  }
}```

More info about this resource (and the log_query section) can be found in our provider docs - https://www.terraform.io/docs/providers/datadog/r/dashboard.html#nested-apm_query-and-log_query-blocks Since this feature has been released I'll go ahead and close this issue, but please do let us know if you hit any further issues with this!
